sklearn/preprocessing/tests/test_data.py

```diff
@@ -851,6 +854,298 @@ def test_robust_scaler_iris_quantiles():
+    assert_array_almost_equal(np.max(X_trans, axis=0), 1.)
+    X_trans_inv = transformer.inverse_transform(X_trans)
+    assert_array_almost_equal(X, X_trans_inv)
+
+
+def test_quantile_transform_check_error():
+    X = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                  [2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                  [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    X = sparse.csc_matrix(X)
+    X_neg = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                      [-2, 4, 0, 0, 6, 8, 0, 10, 0, 0],
+                      [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    X_neg = sparse.csc_matrix(X_neg)
+
+    assert_raises_regex(ValueError, "Invalid value for 'n_quantiles'",
+                        QuantileTransformer(n_quantiles=0).fit, X_neg)
+    assert_raises_regex(ValueError, "Invalid value for 'subsample'",
+                        QuantileTransformer(subsample=0).fit, X_neg)
+    assert_raises_regex(ValueError, "Invalid value for 'smoothing_noise'",
+                        QuantileTransformer(smoothing_noise=0).fit, X_neg)
+
+    transformer = QuantileTransformer(n_quantiles=10)
+    assert_raises_regex(ValueError, "QuantileTransformer only accepts "
+                        "non-negative sparse matrices", transformer.fit, X_neg)
+    transformer.fit(X)
+    assert_raises_regex(ValueError, "QuantileTransformer only accepts "
+                        "non-negative sparse matrices",
+                        transformer.transform, X_neg)
+    assert_raises_regex(ValueError, "QuantileTransformer only accepts "
+                        "non-negative sparse matrices",
+                        transformer.inverse_transform, X_neg)
+
+    X_bad_feat = np.array([[0, 25, 50, 0, 0, 0, 75, 0, 0, 100],
+                           [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]]).T
+    assert_raises_regex(ValueError, "X does not have the same number of "
+                        "feature than the previously fitted data.",
+                        transformer.transform, X_bad_feat)
+    assert_raises_regex(ValueError, "X does not have the same number of "
+                        "feature than the previously fitted data.",
+                        transformer.inverse_transform, X_bad_feat)
+
+    assert_raises_regex(ValueError, "'output_distribution' has to be either"
+                        " 'norm' or 'uniform'. Got rnd instead.",
+                        QuantileTransformer(
+                            n_quantiles=10,
+                            output_distribution='rnd').fit_transform, X)
+    assert_raises_regex(ValueError, "'output_distribution' has to be either"
+                        " 'norm' or 'uniform'. Got rnd instead.",
+                        QuantileTransformer(n_quantiles=10,
+                                            output_distribution='rnd').fit(
+                            X).inverse_transform, X)
+
+
+def test_quantile_transform_ignore_zeros():
```

Comment:
wouldn't this be more cohesive appearing after `test_quantile_transform_sparse_toy`?